hts_expr.c

```diff
@@ -688,6 +688,15 @@ int hts_filter_eval(hts_filter_t *filt,
                     hts_expr_val_t *res) {
     char *end = NULL;
 
+    if (res->s.l != 0 || res->s.m != 0 || res->s.s != NULL) {
+        // As *res is cleared below, it's not safe to call this function
+        // with res->s.s set, as memory would be leaked.  It's also not
+        // possible to know is res was initialised correctly, so in
+        // either case we fail ungracefully.
+        fprintf(stderr, "hts_expr_val_t not initialised (with HTS_EXPR_VAL_INIT) or cleared (by hts_expr_val_free) correctly.  Not safe to continue, sorry.\n");
+        abort();
```

Comment:
Doesn't think break our rule that the library should never just call abort?

It ought to return -1 instead (if indeed we wish to even go down this road).

Reply:
We don't call `abort()` for conditions that could be triggered by dodgy inputs, but we do in places where it would indicate a bug in HTSlib.  In this case the bug would be in the code that calls HTSlib, but it would still be something that needs the attention of the code developer rather than the end user.  I can change it to `hts_log_error` and `return -1` if you'd like though.